tests/virtual_machines_test.go

```diff
@@ -489,6 +489,63 @@ var _ = Describe("Virtual Machines", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("When a VM's NIC is removed and a new VM is created with the same MAC", func() {
+			It("should successfully release the MAC and the new Vm should be created with no errors", func() {
+				err := setRange(rangeStart, rangeEnd)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", ""), newInterface("br2", "")},
+					[]kubevirtv1.Network{newNetwork("br1"), newNetwork("br2")})
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm)
+
+				}, 50*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+				mac1, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+				mac2, err := net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[1].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				updatedVM := *vm.DeepCopy()
+				By("before the NIC is deleted")
+				newVm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", mac1.String())},
+					[]kubevirtv1.Network{newNetwork("br1")})
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), newVm)
```

Comment:
please validate the error here. we need to check that the error is related to the mac allocation and not something else